### ISSUE TITLE
Atomic gets that depend on each other

### ISSUE THREAD

**Issue by refi64** [lucaong/cubdb#27]
TLDR what I want to do is something like this, but atomic:

```elixir
k2 = CubDB.get(db, {:index, k1})
v = CubDB.get(db, {:stuff, k2})
```

In other words, I'm taking the value of one key and using it as part of another key, which doesn't seem to be possible with current APIs. Now for my use case the atomicity isn't *essential*, so I could just use the double gets. However, some support for this would be nice to have regardless...

**Comment by lucaong** [comment]
Hi @refi64 ,
you are right in that, while it is possible to get multiple "independent" entries atomically, there is currently no API to atomically get two entries, with the second key depending on the first. I will think about this use case, it would be great to enable it.

**Comment by lucaong** [comment]
Hi @refi64 , I know this comes quite a long time after this issue was open, but this pull request (currently in draft) should meet your original need by introducing snapshots: https://github.com/lucaong/cubdb/pull/47

**Comment by lucaong** [comment]
This is now in `v2.0.0-rc.1`, and will soon be released as part of `v2.0.0`
